tests/proxy_tests.rb

```diff
@@ -100,6 +100,10 @@ def env_proxy_tests(env)
           connection.data[:proxy][:scheme]
         end
 
+        tests('connection.data[:proxy][:headers]').returns({ 'x-proxy-id': 'abc123' }) do
```

Comment:
These tests are good for ensuring the data mutation into proxy happens as expected (unless we decide not to do that).

I think it would be great to also add a test toward the end of this file, where it uses the `proxy.ru` rackup. These tests actually run a server, which receives proxy stuff and echos it back (so you can be more confident it was sent as expected). I think it would be great to also test the roundtrip on these proxy settings, though this would require updating `proxy.ru` as well. I think maybe we could update it so the response passes back the value of `x-proxy-id` and check the response headers to gain this confidence.

Does that make sense? What do you think?

Reply:
yep! always happy to write more tests 😁 I'll dig into that